api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -412,6 +412,41 @@ public String toString() {
     }
   }
 
+  public static class VariantType implements Type {
```

Comment:
@aihuaxu, could you also implement the same Java serialization that we use for `PrimitiveType`? The existing `PrimitiveHolder` should work:

```
    Object writeReplace() throws ObjectStreamException {
      return new PrimitiveHolder(toString());
    }
```